src/shared/sleep-config.c

```diff
@@ -31,31 +31,53 @@
 #include "string-util.h"
 #include "strv.h"
 
-int parse_sleep_config(const char *verb, bool *ret_allow, char ***ret_modes, char ***ret_states, usec_t *ret_delay) {
-        int allow_suspend = -1, allow_hibernate = -1,
-            allow_s2h = -1, allow_hybrid_sleep = -1;
-        bool allow;
-        _cleanup_strv_free_ char
-                **suspend_mode = NULL, **suspend_state = NULL,
-                **hibernate_mode = NULL, **hibernate_state = NULL,
-                **hybrid_mode = NULL, **hybrid_state = NULL;
-        _cleanup_strv_free_ char **modes, **states; /* always initialized below */
-        usec_t delay = 180 * USEC_PER_MINUTE;
+SleepConfig *sleep_config_new(void) {
+        _cleanup_(free_sleep_configp) SleepConfig *sc = NULL;
+
+        sc = new(SleepConfig, 1);
+        if (!sc)
+                return NULL;
+
+        *sc = (SleepConfig) {
+                .allow_suspend       = -1,
+                .allow_hibernate     = -1,
+                .allow_s2h           = -1,
+                .allow_hybrid_sleep  = -1,
+
+                .suspend_modes       =  NULL,
+                .suspend_states      =  NULL,
+                .hibernate_modes     =  NULL,
+                .hibernate_states    =  NULL,
+                .hybrid_modes        =  NULL,
+                .hybrid_states       =  NULL,
+
+                .hibernate_delay_sec = 0
```

Comment:
No need to initialize anything that is NULL or 0. Just remove that.

But please keep the trailing comma after last initialization (i.e. the one in `.allow_hybrid_sleep = -1,`).